presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -499,9 +503,12 @@ protected void setup(Binder binder)
         binder.bind(NodeVersion.class).toInstance(nodeVersion);
 
         // presto announcement
+        checkArgument(!(serverConfig.isResourceManager() && serverConfig.isCoordinator()),
```

Comment:
This check should be done at the top if we are not allowing RM and coordinator to be enabled on single machine.